README.md

```diff
@@ -22,12 +27,370 @@ Download the collection and extract it in the `data` directory, under a `collect
 ...
 ```
 
-We assume you're in a correct new *Virtual Environnement*, with `python` launching a Python 3.7+.
+**Caution:** We assume you're in a correct new *Virtual Environnement*, with `python` and `pip` refering to Python 3.7+.
 
-You can run `make install`.
+To install the modules, you can simply run `make install`.
 
 Or Install the necessary modules : `pip install -r requirements.txt`, and the minimal *nltk* modules : `python -m nltk.downloader popular`.
 
+We specifically relie on two modules for this project :
+
+- `nltk`, a natural language processing toolkit, used for tokenization and lemmatization. We use *wordnet*, so you'll need to download additional data.
+- `ttable`, a library to work with Boolean expressionss
+
+## Usage
+
+TODO, with interface.py
+
+## Preprocessing
+
+### Files loading
+
+We simply open all files in the `collection` folder, loading the content in memory into a dictionnary.
+
+Since there is almost 100 000 documents, it takes aroud **15 sec** to load all the files. To avoid waiting for this operation, we save the corpus once fully loaded as a binary with `pickle` at `data/corpus.pkl`. This file takes only *161 Mo* and is loaded in **0.38 sec**.
+
+### Tokenization
+
+The data set is already tokenized, but we still this process for queries.
+We use the tokenization provided with `nltk`, which is more precise than a simple `split` on spaces, for numbers or `'s` for example.
+
+### Stop words
+
+We remove stop words before *lemmatization* but also after. The idea is that we want to remove all obvious useless words from the dataset before the quite *costly* lemmatization, but also remove some synonyms of stop words find with the *lemmatization*. Since the stop word removal is quite negligeable in term of computing time for the whole index creation, we think it's not a bad choice.
+
+### Lemmatization
+
+For a better quality of lemmatization, we use the *context* of a token in a sentence, with the `pos_tag` (*Part-Of-Speach* tag) function in `nltk`, which can indicate the grammatical use of a word : *fly* can be either a verb or a noun. The *WordNet* lemmatizer can take as argument this categorisation to give a precise result.
+
+However, since this process quite is long, so we also used the simpler version with the `SnowballStemmer` from `nltk`, which is faster.
```

Comment:
It must be clear that the snowball Stemmer is only a stemming process, but it is so effective that applying lemmatization + stemming with this stemmer yields almost always the same results as the stemming alone. Also specify that whichever process is chosen must also be used for the query.